more

david-987 opened the pull request.

david-987 committed:
more

david-987 commented on src/client/tsconfig.json:
```diff
@@ -0,0 +1,22 @@
+        "module": "esnext",
+        "isolatedModules": true,
+        "jsx": "preserve",
+        "baseUrl": ".",
```
the baseUrl seems important here, won't resolve `'client/'` imports otherwise within the client project

amcasey replied:
Note that I didn't actually remove the `baseUrl` in my PR - I lifted into a settings file at a higher level.  You absolutely do need a `baseUrl` if you're going to use path mappings.  My recommendation is basically to keep them together so it's clear where the path mappings resolve to.

david-987 replied:
What I meant here is that lifting it to a higher level caused my client project not to resolve `client/*` imports. Adding it directly in this file fixed it, but I might've done other stuff wrong. Gonna give this another go soon

amcasey replied:
And the reason not to lift the path mapping up as well is that only `client` imports from `client/*`?

david-987 replied:
correct, only client imports from client/*

david-987 committed:
fix name

david-987 commented on src/server/tsconfig.json:
```diff
@@ -1,5 +1,6 @@
 {
     "extends": "../../tsconfig.settings.json",
+    "include": ["src/**/*"],
```
would not work without this include

amcasey replied:
Yeah, as I mentioned on the other PR, that was an error on my part - sorry about that.

amcasey replied:
Wait, even with `rootDir: src`?

david-987 commented:
@amcasey Tried to populate the repo with more items such as our esm builds. As I mentioned in another comment, trying to implement the tweaks you suggested to my prod repo broke a lot, and spent a few hours playing whack-a-mole before giving up.
Our server build is performed `tsc -b tsconfig.json tsconfig.esm.json` and fails horribly during the esm step, couple hundred errors complaining about 
`File 'C:/Users/David/Documents/...../util/src/core/ArrayUtil/ArrayUtil.ts' is not under 'rootDir' 'C:/Users/David/Documents/...../src/package/business/src'. 'rootDir' is expected to contain all source files.`

`File 'C:/Users/David/Documents/....../src/package/util/src/core/client.ts' is not listed within the file list of project 'C:/Users/David/Documents/....../src/package/business/tsconfig.esm.json'. Projects must list all files or use an 'include' pattern.`

Tried by best to reproduce this here but no luck, probably missing something... this stuff gets really messy and it's hard to figure out the diffs.

amcasey commented on src/packages/user2/tsconfig.esm.json:
```diff
@@ -0,0 +1,7 @@
+{
```
It looks like you might be missing `references`?  They're not inherited via `extends` and, even if they were, you want the `esm` version, right?

david-987 replied:
Yes, forgot. Thanks for this comment, ended up spotting some issues in my real repo due to this.

amcasey commented on jsconfig.json:
```diff
@@ -0,0 +1,8 @@
+{
```
What's this file for?

amcasey replied:
I still don't understand what this file is for.

david-987 replied:
sorry these are old files, should be deleted

david-987 commented:
@amcasey so to give an update, moving `"baseUrl": ".",` up to root revealed a bunch of TS errors that were not appearing before (still don't appear in vscode, only during a `tsc -b` run output). Fixed most but some of them are a bit odd so i've `// @ts-ignore`'d them for now. Also fixed the esm config files which helped. It appears to have fixed the OOM issue in my real repo.

Some of the odd ones in the `/server` project:

```
src/productA/controller.ts:41:17 - error TS2322: Type 'import("C:/Users/David/Documents/<omitted>/src/package/util/dist/core/Optional").default<import("C:/Users/David/Documents/<omitted>/src/package/business/node_modules/@packages/geo/dist/location/models/LatLng").default>' 
  is not assignable to type 
'import("C:/Users/David/Documents/<omitted>/src/package/business/node_modules/@packages/geo/node_modules/@packages/util/dist/core/Optional").default<import("C:/Users/David/Documents/<omitted>/src/package/business/node_modules/@packages/geo/dist/location/models/LatLng").default>'.
  Types have separate declarations of a private property 'value'.
```
What am curious about is why it would go through the business project and then the geo project to import `Optional` instead of getting it directly from util.
Why reference the business project at all? It doesn't seem relevant for the code.
As a side note they all reference the same Optional and LatLng classes, yet it seems to think they're different.

david-987 commented:
> @amcasey so to give an update, moving `"baseUrl": ".",` up to root revealed a bunch of TS errors that were not appearing before (still don't appear in vscode, only during a `tsc -b` run output). Fixed most but some of them are a bit odd so i've `// @ts-ignore`'d them for now. Also fixed the esm config files which helped. It appears to have fixed the OOM issue in my real repo.
> 
> Some of the odd ones in the `/server` project:
> 
> ```
> src/productA/controller.ts:41:17 - error TS2322: Type 'import("C:/Users/David/Documents/<omitted>/src/package/util/dist/core/Optional").default<import("C:/Users/David/Documents/<omitted>/src/package/business/node_modules/@packages/geo/dist/location/models/LatLng").default>' 
>   is not assignable to type 
> 'import("C:/Users/David/Documents/<omitted>/src/package/business/node_modules/@packages/geo/node_modules/@packages/util/dist/core/Optional").default<import("C:/Users/David/Documents/<omitted>/src/package/business/node_modules/@packages/geo/dist/location/models/LatLng").default>'.
>   Types have separate declarations of a private property 'value'.
> ```
> 
> What am curious about is why it would go through the business project and then the geo project to import `Optional` instead of getting it directly from util. Why reference the business project at all? It doesn't seem relevant for the code. As a side note they all reference the same Optional and LatLng classes, yet it seems to think they're different.

Made more tweaks and the odd type errors went away, am not sure what exactly fixed it.
I'll update this rep soon to mirror my real project and see if you have any additional feedback.

david-987 committed:
more

david-987 commented:
@amcasey updated the repo, when you get a chance let me know if anything looks off!

amcasey reviewed (commented):
I'm confused by the jsconfig files.  What are they for?

david-987 committed:
delete old files

david-987 committed:
manual reference packages

david-987 committed:
simplify
